service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.service.digitaltwin;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.authentication.SasTokenProvider;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.authentication.ServiceClientCredentialsProvider;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.authentication.ServiceConnectionString;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.authentication.ServiceConnectionStringParser;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.customized.DigitalTwinGetHeaders;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.customized.DigitalTwinUpdateHeaders;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.implementation.DigitalTwinsImpl;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.implementation.IotHubGatewayServiceAPIsImpl;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.DigitalTwins;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.models.DigitalTwinInvokeRootLevelCommandHeaders;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.serialization.DeserializationHelpers;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.serialization.DigitalTwinStringSerializer;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.models.*;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+import com.microsoft.rest.*;
+import com.microsoft.rest.serializer.JacksonAdapter;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.microsoft.azure.sdk.iot.service.digitaltwin.helpers.Tools.*;
+
+public class DigitalTwinAsyncClient {
+    private DigitalTwinsImpl digitalTwin;
```

Comment:
can we be consistent with the track 2 and the C# sdk and call this _protocolLayer?

Reply:
done